sdk/clientcore/core/src/main/java/io/clientcore/core/http/pipeline/HttpPipelineOrder.java

```diff
@@ -0,0 +1,164 @@
+     * <p>
+     * If multiple {@link HttpPipelinePolicy policies} are added with this order, they will be executed in the order
+     * they were added between the {@link HttpRetryPolicy} and {@link HttpCredentialPolicy}. Or, visually:
+     * <pre>
+     *     +-----------------------+
+     *     |     Retry Policy      |
+     *     +-----------------------+
+     *     |   1st added Between   |
+     *     +-----------------------+
+     *     |   2nd added Between   |
+     *     +-----------------------+
+     *     |   3rd added Between   |
+     *     +-----------------------+
+     *     | Authentication Policy |
+     *     +-----------------------+
+     * </pre>
+     */
+    public static final HttpPipelineOrder BETWEEN_RETRY_AND_AUTHENTICATION = new HttpPipelineOrder(5000);
+
+    /**
+     * The policy will be position after the {@link HttpCredentialPolicy} and before the
+     * {@link HttpInstrumentationPolicy}.
+     * <p>
+     * If multiple {@link HttpPipelinePolicy policies} are added with this order, they will be executed in the order
+     * they were added between the {@link HttpCredentialPolicy} and {@link HttpInstrumentationPolicy}. Or, visually:
+     * <pre>
+     *     +------------------------+
+     *     | Authentication Policy  |
+     *     +------------------------+
+     *     |   1st added Between    |
+     *     +------------------------+
+     *     |   2nd added Between    |
+     *     +------------------------+
+     *     |   3rd added Between    |
+     *     +------------------------+
+     *     | Instrumentation Policy |
+     *     +------------------------+
+     * </pre>
+     */
+    public static final HttpPipelineOrder BETWEEN_AUTHENTICATION_AND_INSTRUMENTATION = new HttpPipelineOrder(7000);
+
+    /**
+     * The policy will be position after the {@link HttpInstrumentationPolicy}.
+     * <p>
+     * If multiple {@link HttpPipelinePolicy policies} are added with this order, they will be executed in the order
+     * they were added after {@link HttpInstrumentationPolicy}. Or, visually:
+     * <pre>
+     *     +------------------------+
+     *     | Instrumentation Policy |
+     *     +------------------------+
+     *     |    1st added After     |
+     *     +------------------------+
+     *     |    2nd added After     |
+     *     +------------------------+
+     *     |    3rd added After     |
+     *     +------------------------+
+     * </pre>
+     */
+    public static final HttpPipelineOrder AFTER_INSTRUMENTATION = new HttpPipelineOrder(9000);
+
+    // Package-private HttpPipelineOrder constants for pillar policies.
+    static final HttpPipelineOrder REDIRECT = new HttpPipelineOrder(2000);
+    static final HttpPipelineOrder RETRY = new HttpPipelineOrder(4000);
+    static final HttpPipelineOrder AUTHENTICATION = new HttpPipelineOrder(6000);
+    static final HttpPipelineOrder INSTRUMENTATION = new HttpPipelineOrder(8000);
```

Comment:
9000 policies should be enough for everyone! let's see if this ages well

Reply:
![image](https://github.com/user-attachments/assets/4b4c290f-0d84-4196-b8d4-1ec47457c01c)